client/pool_test.go

```diff
@@ -0,0 +1,132 @@
+package client
+
+import (
+	"fmt"
+	"gopkg.in/jarcoal/httpmock.v1"
+	"net/http"
+	"testing"
+	"time"
+)
+
+func NewTestPool() *ClientPool {
+	p, _ := NewClientPool([]string{"http://localhost:8332", "http://localhost:8336"}, nil)
+	p.client = &http.Client{}
+	p.Start()
+	return p
+}
+
+func TestServerRotation(t *testing.T) {
+	setup()
+	defer teardown()
+
+	var (
+		p          = NewTestPool()
+		testPath   = fmt.Sprintf("http://%s/tx/1be612e4f2b79af279e0b307337924072b819b3aca09fcb20370dd9492b83428", p.apiUrl.Host)
+		testPath2  = fmt.Sprintf("http://%s/tx/1be612e4f2b79af279e0b307337924072b819b3aca09fcb20370dd9492b83428", "localhost:8336")
+		expectedTx = TestTx
+	)
+
+	// Test valid response from first server
+	response, err := httpmock.NewJsonResponse(http.StatusOK, expectedTx)
+	if err != nil {
+		t.Error(err)
+	}
+
+	httpmock.RegisterResponder(http.MethodGet, testPath,
+		func(req *http.Request) (*http.Response, error) {
+			return response, nil
+		},
+	)
+
+	tx, err := p.GetTransaction("1be612e4f2b79af279e0b307337924072b819b3aca09fcb20370dd9492b83428")
+	if err != nil {
+		t.Error(err)
+	}
+	validateTransaction(*tx, expectedTx, t)
+
+	// Test invalid response, server rotation, then valid response from second server
+	response1, err := httpmock.NewJsonResponse(http.StatusInternalServerError, expectedTx)
+	if err != nil {
+		t.Error(err)
+	}
+
+	response2, err := httpmock.NewJsonResponse(http.StatusOK, expectedTx)
+	if err != nil {
+		t.Error(err)
+	}
+
+	httpmock.RegisterResponder(http.MethodGet, testPath,
+		func(req *http.Request) (*http.Response, error) {
+			return response1, nil
+		},
+	)
+
+	httpmock.RegisterResponder(http.MethodGet, testPath2,
+		func(req *http.Request) (*http.Response, error) {
+			return response2, nil
+		},
+	)
+
+	tx, err = p.GetTransaction("1be612e4f2b79af279e0b307337924072b819b3aca09fcb20370dd9492b83428")
```

Comment:
Using a variable instead of a literal for the txid shows they are implicitly equivalent without having to verify it visually. Consider changing it?